packages/scene-composer/src/components/three-fiber/AnchorComponent/index.tsx

```diff
@@ -12,15 +11,13 @@ interface IAnchorComponentProps {
 }
 
 const AnchorComponent: React.FC<IAnchorComponentProps> = ({ node, component }: IAnchorComponentProps) => {
-  const sceneComposerId = useSceneComposerId();
-  const rule = useStore(sceneComposerId)((state) => state.getSceneRuleMapById(component.ruleBasedMapId));
   return (
     <group name={getComponentGroupName(node.ref, 'TAG')}>
       <AnchorWidget
         node={node}
         defaultIcon={component.icon ?? DefaultAnchorStatus.Info}
         navLink={component.navLink}
-        rule={rule}
+        ruleBasedMapId={component.ruleBasedMapId}
```

Comment:
What's the reason for not fetching it from the store? Are we not storing these maps in the store?

Reply:
the `useRuleResult` now used by AnchorWidget will fetch the rule by id